sql/core/src/test/scala/org/apache/spark/sql/streaming/StreamSuite.scala

```diff
@@ -553,6 +554,32 @@ class StreamSuite extends StreamTest {
     }
   }
 
+  test("SPARK-19909: if the checkpoint location is not set and the default filesystem " +
+    "is different from the java.io.tmp one an AnalysisException should be thrown") {
+
+    val defaultFS = spark.conf.getOption(FileSystem.FS_DEFAULT_NAME_KEY)
+    spark.conf.set(FileSystem.FS_DEFAULT_NAME_KEY, "hdfs:///")
+
+    val input = MemoryStream[Int]
+    var streamingQuery: StreamingQuery = null
+    input.addData(1)
+    try {
+      intercept[AnalysisException](
+        streamingQuery = input.toDF().writeStream.format("console").start()
+      )
+    } finally {
+      if (streamingQuery ne null) {
```

Comment:
Can you please change to `streamingQuery != null`? Spark seldom uses `ne`, we'd better follow the convention.